Source/Softswitch/inc/poets_msg.h

```diff
@@ -74,39 +118,70 @@ typedef struct poets_message
     // Further flits contain data.
 } P_Msg_t;
 
-typedef struct p_super_msg_header
+typedef struct poets_log_message_payload
 {
-    // flit 0
-    uint32_t sourceDeviceAddr; // issuing device in the Application
-    uint16_t command;          // commands might be split into (fixed) system commands and user-defined commands
-    uint16_t destPin;          // destination pin on the Supervisor
-    uint32_t cmdLenBytes;      // length of the entire command (including all headers)
-    uint32_t seq;              // sequence number of this sub-message
+    uint8_t seq;
+    uint8_t payload[P_MSG_MAX_SIZE-(sizeof(P_Msg_Hdr_t)+sizeof(uint8_t))];
+} P_Log_Msg_Pyld_t;
 
-    // further flits contain data - which could be arguments, packets, data dumps, or just about anything else. 
-} P_Sup_Hdr_t;
+typedef struct poets_instr_message_payload
+{
+    // First Flit
+    uint32_t cIDX;
+    uint32_t cycles;
+    
+    // Second Flit
+    uint32_t rxCnt;
+    uint32_t rxHanCnt;
+    uint32_t txCnt;
+    uint32_t supCnt;
+    
+    // Third Flit
+    uint32_t txHanCnt;
+    uint32_t idleCnt;
+    uint32_t idleHanCnt;
+    uint32_t blockCnt;
+    
+#if TinselEnablePerfCount == true   
+    // Fourth Flit
+    uint32_t missCount;         // Cache miss count
+    uint32_t hitCount;          // Cache hit count
+    uint32_t writebackCount;    // Cache writeback count
+    uint32_t CPUIdleCount;      // CPU idle-cycle count (lower 32 bits)
+    
+    // 0-bits free
+#endif 
+    
+} P_Instr_Msg_Pyld_t;
 
-typedef struct p_super_message
+
+typedef struct poets_supervisor_message
 {
-    P_Sup_Hdr_t header; // a Supervisor message always has an appropriate header
-    uint8_t data[P_MSG_MAX_SIZE-sizeof(P_Sup_Hdr_t)]; // and it might have some sort of data
-} P_Sup_Msg_t;
+    uint32_t hwAddr;    // The destination HW address
+    uint32_t len;       // Number of bytes for the P_Msg_t
+    P_Msg_t msg;        // The POETs fabric message
+} P_Super_Msg_t;
```

Comment:
:black_flag:

Reply:
What does a black flag mean? :P

Reply:
So, I have remembered why this is still about - the "new" header does not include the Hardware address. This struct is for use in the supervisor to keep the destination hardware address with the message to be sent.